stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -35,6 +37,39 @@ function SparseMatrixCSC(m::Integer, n::Integer, colptr::Vector, rowval::Vector,
     SparseMatrixCSC{Tv,Ti}(m, n, colptr, rowval, nzval)
 end
 
+ @noinline function sparse_check_Ti(m::Integer, n::Integer, Ti::Type)
+        @noinline throwsz(str, lbl, k) =
+            throw(ArgumentError("number of $str ($lbl) must be ≥ 0, got $k"))
+        @noinline throwTi(str, lbl, k) =
+            throw(ArgumentError("$str ($lbl = $k) does not fit in Ti = $(Ti)"))
+        m < 0 && throwsz("rows", 'm', m)
+        n < 0 && throwsz("columns", 'n', n)
+        !isbitstype(Ti) || m ≤ typemax(Ti) || throwTi("number of rows", "m", m)
```

Comment:
Do we really need to assert `!isbitstype`?  

```
julia> isbitstype(BigInt)
false

julia> sparse([big(1)], [big(1)], [1], big(50), big(50))
50×50 SparseMatrixCSC{Int64,BigInt} with 1 stored entry:
  [1, 1]  =  1
```

I mean, its performance will likely not be great, but this is the sort of breaking change that we don't want to introduce just for the sake of being opinionated on how folks use sparse matrices.

Reply:
I an declare myself not guilty for being opinionated. :-) 

The purpose of the `!isbitstype` was to disable the check for `BigInt`, not to disallow `BigInt` for `Ti`. `typemax(BigInt)` does not return any value to compare with.

Reply:
🤦‍♂️ 

Of course, this is an `||` chain and not an `&&` one.  :+1: